api/src/users/users.service.ts

```diff
@@ -258,4 +284,40 @@ export class UsersService {
       throw new InternalServerErrorException();
     }
   }
+
+  async updateUser(userId: number, updateUsersDto: UpdateUsersDto) {
+    if (updateUsersDto.bio) {
+      try {
+        const result = await db
+        .updateTable('user')
+        .set('bio', updateUsersDto.bio)
+        .where('id', '=', userId)
+        .executeTakeFirst()
+      } catch (error) {
+        throw new InternalServerErrorException();        
+      }
+    }
+    if (updateUsersDto.firstname) {
+      try {
+        const result = await db
+        .updateTable('user')
+        .set('firstname', updateUsersDto.firstname)
+        .where('id', '=', userId)
+        .executeTakeFirst()
+      } catch (error) {
+        throw new InternalServerErrorException();        
+      }
+    }
+    if (updateUsersDto.lastname) {
+      try {
+        const result = await db
+        .updateTable('user')
+        .set('lastname', updateUsersDto.lastname)
+        .where('id', '=', userId)
+        .executeTakeFirst()
+      } catch (error) {
+        throw new InternalServerErrorException();        
+      }
+    }
+  }
```

Comment:
Update the table once. If necessary, create your updated object by fetching the user first, and then update. Perhaps, the `UpdateUsersDto` should have optional attributes. This way you could update like so: `.set({...updateUsersDto})`

Reply:
Fix f091cbb